server/data/data-library/src/main/java/org/apache/james/user/lib/AbstractUsersRepository.java

```diff
@@ -73,7 +74,7 @@ protected void isValidUsername(String username) throws UsersRepositoryException
             if (i == -1) {
                 throw new UsersRepositoryException("Given Username needs to contain a @domainpart");
             } else {
-                String domain = username.substring(i + 1);
+                Domain domain = Domain.of(username.substring(i + 1));
```

Comment:
Some parsing lost in the wild. 

Maybe user.getDomain works?

Reply:
changed in another commit

Reply:
where?